google-cloud-datastore/lib/google/cloud/datastore/aggregate_query.rb

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/datastore/v1"
+
+module Google
+  module Cloud
+    module Datastore
+      ##
+      # # AggregateQuery
+      #
+      # An aggregate query can be used to fetch aggregate values (ex: count) for a query
+      #
+      # @example
+      #   require "google/cloud/datastore"
+      #
+      #   datastore = Google::Cloud::Datastore.new
+      #
+      #   query = Google::Cloud::Datastore::Query.new
+      #   query.kind("Task").
+      #     where("done", "=", false)
+      #
+      #   Create an aggregate query
+      #   aggregate_query = query.aggregate_query
+      #                          .add_count
+      #
+      #   aggregate_query_results = dataset.run_aggregation aggregate_query
+      #   puts aggregate_query_results.get('count')
+      #
+      # @example Alias an aggregate query
+      #   require "google/cloud/datastore"
+      #
+      #   datastore = Google::Cloud::Datastore.new
+      #
+      #   query = Google::Cloud::Datastore::Query.new
+      #   query.kind("Task").
+      #     where("done", "=", false)
+      #
+      #   Create an aggregate query
+      #   aggregate_query = query.aggregate_query
+      #                          .add_count aggregate_alias: 'total'
+      #
+      #   aggregate_query_results = dataset.run_aggregation aggregate_query
+      #   puts aggregate_query_results.get('total')
+      #
+      class AggregateQuery
+        ##
+        # @private The Google::Cloud::Datastore::V1::Query object.
+        attr_reader :query
+
+        ##
+        # @private Array of Google::Cloud::Datastore::V1::AggregationQuery::Aggregation objects
+        attr_reader :aggregates
+
+        ##
+        # @private Creates a new AggregateQuery
+        def initialize query
+          @query = query
+          @aggregates = []
+        end
+
+        ##
+        # Adds a count aggregate.
+        #
+        # @param [aggregate_alias] Alias to refer to the aggregate. Optional
+        #
+        # @return [AggregateQuery] The modified aggregate query object with the added count aggregate.
+        #
+        # @example
+        #   require "google/cloud/datastore"
+        #
+        #   datastore = Google::Cloud::Datastore.new
+        #
+        #   query = Google::Cloud::Datastore::Query.new
+        #   query.kind("Task").
+        #     where("done", "=", false)
```

Comment:
Use the same style for examples that we do in code. In this case, the period goes on the next line and should be aligned with the period above it, like you do in lines 91-92. Same for the corresponding lines in the next example below, and examples in subsequent files.

Reply:
Actually, the existing examples in Datastore uses the convention of trailing periods, so I followed the same convention to keep it consistent. Do you prefer I follow the other convention in this PR?

Reply:
The convention enforced by our Rubocop configuration is leading-period. We should follow that for all new examples. (We should also update older examples to follow the correct convention, but that's a lower priority task and should be done separately.)

At the very least, it looks sloppy if conflicting styles appear just a few lines apart in the same example. We should change either one or the other to be consistent. And if we're going to change one, might as well change it to be the same convention we use in our code.

Reply:
I changed the examples to have leading periods. I'll do a follow up PR to change the rest of the datastore examples to follow this convention.